examples/express/index.ts

```diff
@@ -0,0 +1,135 @@
+import express, { Router } from 'express'
+import SgidClient from '@opengovsg/sgid-client'
+import * as dotenv from 'dotenv'
+import crypto from 'crypto'
+import cookieParser from 'cookie-parser'
+import { fetchStaticFiles } from './helpers'
+import open from 'open'
+
+dotenv.config()
+
+const PORT = 5000
+
+const sgid = new SgidClient({
+  clientId: String(process.env.SGID_CLIENT_ID),
+  clientSecret: String(process.env.SGID_CLIENT_SECRET),
+  privateKey: String(process.env.SGID_PRIVATE_KEY),
+  redirectUri: `http://localhost:${PORT}/api/callback`,
+})
+
+const app = express()
+
+const apiRouter = Router()
+
+const SESSION_COOKIE_NAME = 'exampleAppSession'
+const SESSION_COOKIE_OPTIONS = {
+  httpOnly: true,
+}
+
+type SessionData = Record<
+  string,
+  | {
+      nonce?: string
+      // Store state as search params to easily stringify key-value pairs
+      state: URLSearchParams
+      accessToken?: string
+    }
+  | undefined
+>
+
+/**
+ * In-memory store for session data.
+ * In a real application, this would be a database.
+ */
+const sessionData: SessionData = {}
+
+apiRouter.get('/auth-url', (req, res) => {
+  const iceCreamSelection = String(req.query.icecream)
+  const sessionId = crypto.randomUUID()
+  // Use search params to store state so other key-value pairs
+  // can be added easily
+  const state = new URLSearchParams({
+    icecream: iceCreamSelection,
+  })
+  const { url, nonce } = sgid.authorizationUrl(
+    // We pass the user's ice cream preference as the state,
+    // so after they log in, we can display it together with the
+    // other user info.
+    state.toString(),
+    // Scopes that all sgID relying parties can access by default
+    ['openid', 'myinfo.name'],
+  )
+  sessionData[sessionId] = {
+    state,
+    nonce,
+  }
+  return res
+    .cookie(SESSION_COOKIE_NAME, sessionId, SESSION_COOKIE_OPTIONS)
+    .json({ url })
```

Comment:
is there a reason that we are returning the `url` to the FE and letting the FE redirect instead of directly redirecting from this endpoint?
e.g.
```
res.redirect(url)
```

Reply:
The reason for this is that the redirect request will be blocked by CORS because `api.id.gov.sg` does not have the proper CORS headers set. This happens because this is a response from the initial request by the browser